alicloud/resource_alicloud_network_interface.go

```diff
@@ -0,0 +1,342 @@
+		Read:   resourceAliyunNetworkInterfaceRead,
+		Update: resourceAliyunNetworkInterfaceUpdate,
+		Delete: resourceAliyunNetworkInterfaceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"security_groups": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				MinItems: 1,
+			},
+			"private_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"private_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				MaxItems: 10,
+			},
+			"private_ips_count": &schema.Schema{
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validatePrivateIpsCount,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAliyunNetworkInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+	args := ecs.CreateCreateNetworkInterfaceRequest()
+
+	args.VSwitchId = d.Get("vswitch_id").(string)
+	groups := d.Get("security_groups").(*schema.Set).List()
+
+	args.SecurityGroupId = groups[0].(string)
+
+	if primaryIpAddress, ok := d.GetOk("private_ip"); ok {
+		args.PrimaryIpAddress = primaryIpAddress.(string)
+	}
+
+	if name, ok := d.GetOk("name"); ok {
+		args.NetworkInterfaceName = name.(string)
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		args.Description = description.(string)
+	}
+
+	raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.CreateNetworkInterface(args)
+	})
+	if err != nil {
+		return fmt.Errorf("Create NetworkInterface faield, %#v", err)
+	}
+
+	resp := raw.(*ecs.CreateNetworkInterfaceResponse)
+	d.SetId(resp.NetworkInterfaceId)
+
+	if err := ecsService.WaitForEcsNetworkInterface(d.Id(), Available, DefaultTimeout); err != nil {
+		return fmt.Errorf("Wait NetwortInterface(%s) to be avaialbe failed, %#v", d.Id(), err)
+	}
+
+	return resourceAliyunNetworkInterfaceUpdate(d, meta)
+}
+
+func resourceAliyunNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+
+	eni, err := ecsService.DescribeNetworkInterfaceById("", d.Id())
+	if err != nil {
+		return fmt.Errorf("Describe NetworkInterface(%s) failed, %#v", d.Id(), err)
```

Comment:
If there is no any enis, please return nil

Reply:
Got it